src/main/java/com/odysseusinc/arachne/datanode/service/client/atlas/AtlasAuthRequestInterceptor.java

```diff
@@ -71,8 +83,24 @@ private String authenticate() throws UnsupportedEncodingException {
                 case LDAP:
                     result = loginClient.loginLdap(username, password);
                     break;
+                case ACCESS_TOKEN:
+                    result = generateJWTToken(serviceId, keyfile);
+                    break;
             }
         }
         return result;
     }
+
+    private String generateJWTToken(String serviceId, String keyfile) {
+
+        try(InputStream in = new ByteArrayInputStream(keyfile.getBytes())) {
+            GoogleCredentials credentials = ServiceAccountCredentials.fromStream(in)
+                    .createScoped(Collections.singletonList("https://www.googleapis.com/auth/userinfo.email"));
+            credentials.refreshIfExpired();
```

Comment:
I think it is better to extract "https://www.googleapis.com/auth/userinfo.email" value as constant